doc/tensor.md

```diff
@@ -1327,36 +1331,51 @@ x:indexFill(2,torch.LongTensor{4,2},-10)
 ```
 
 <a name="torch.Tensor.gather"/>
-### [Tensor] gather(src, dim, index) ###
+### [Tensor] torch.gather(src, dim, index) ###
 
-Reads values from `src` at the specified indices into the output tensor, selecting a fixed number of values in each row of `src` along the specified dimension.
-The values of `index` must be between 1 and `src->size(dim)` and all values in a row along the specified dimension must be unique.
-All tensors have the same dimensionality and both `index` and the output tensor are of the same size at the specified dimension.
+Creates a new `Tensor` from the `src` tensor by gathering a number of values from each row or column of `src`. 
+The parameter `dim` specifies whether to gather from rows (`dim = 2`) or columns (`dim = 1`) \[or, more generally, the `dim`th dimesion's rows
```

Comment:
I think I'd rather talk about the general case first, and then use 2D as an example. Wdyt?
Is there a general term for "rows" in the case of more than 2 dimensions? We're essentially talking about all vectors along the specified dimension, aren't we?